pow-migration/src/monitor/mod.rs

```diff
@@ -111,7 +127,16 @@ pub async fn check_validators_ready(
             let filtered_txns: Vec<TransactionDetails> = transactions
                 .into_iter()
                 .filter(|txn| {
-                    // Here we filter by the readiness criteria, TBD
+                    // Check if the txn contains extra data and matches our genesis block hash
+                    if let Some(txn_data) = &txn.data {
```

Comment:
I'd suggest to factorize the txn data check in a function to avoid code duplication